src/replication.c

```diff
@@ -3740,7 +3740,6 @@ void syncWithPrimary(connection *conn) {
     /* Fall through to regular error handling */
 
 error:
-    if (dfd != -1) close(dfd);
```

Comment:
`dfd` is only used in this code:

```C
    /* Prepare a suitable temp file for bulk transfer */
    if (!useDisklessLoad()) {
        while (maxtries--) {
            snprintf(tmpfile, 256, "temp-%d.%ld.rdb", (int)server.unixtime, (long int)getpid());
            dfd = open(tmpfile, O_CREAT | O_WRONLY | O_EXCL, 0644);
            if (dfd != -1) break;
            sleep(1);
        }
        if (dfd == -1) {
            serverLog(LL_WARNING, "Opening the temp file needed for PRIMARY <-> REPLICA synchronization: %s",
                      strerror(errno));
            goto error;
        }
        server.repl_transfer_tmpfile = zstrdup(tmpfile);
        server.repl_transfer_fd = dfd;
    }
```

Let's move the declaration of `dfd` to this small scope to avoid the confusion.